Large improvements and bugfixes

bnm12 opened the pull request:
This PR integrates all the work I've been doing on my fork the past few days

Can be seen live here https://rpg-cards.bnm12.dk/

## Bug fixes
- Fixed a regression where the card preview scaled dynamically on the page rather than keeping the correct size making the layout impossible to trust
- Fixed a bug where cards in the output on pages that weren't full didn't align correctly (due to incorrectly implemented margin)
- Fixed a bug where card backs in the output on pages that weren't full didn't align correctly if you used the background bleed function (on by default) (feature wasn't implemented fully)
- Bootstrap didn't load on the output page causing issues with card content being larger/different between preview and output (normalize css had a significant impact)

## Improvements
- Added JSDoc comments to all the card action handlers
- Added some more sample cards (needs some work) to show some of the newer stuff
- Load Bootstrap on the output page to align the look and styling between preview and output

## New features
- Deploy using github pages rather than Vercel (I didn't want to set it up and you seem to have issues deploying with it currently)
- New card action `sr6spell | Range | Type | Duration | Drain | Damage` that shows a formatted block for shadowrun 6e spells
- New card action `rawhtml | html` which renders raw html. I had issues with the `text` action wrapping block elements in `<p>` tags
- New card actions for table layouts (can replace a lot of the system specific blocks)
    - `table_start` starts a table section
    - `table_head | heading1 | heading2 | heading3 | … | headingN` renders a bold table row for headings
    - `table_row | value1 | value2 | value3 | … | valueN` renders a regular table row (you can have multiple of these)
    - `table_end` ends a table section
- New card action `italic | text` which makes text italic (not everyone knows html)
- New card actions for generic pills - similar to p2e_trait but more flexible and less system specific
    - `pills_start` starts a pill section
    - `pill | text | html-color` shows a pill with the chosen color and text
    - `pills_end` ends a pill section
- New buttons that show all available card actions with tooltips and inserts them in the editor with syntax when pressed
    - This makes the tool more self-documenting and is based on the new JSDoc comments
- New documentation.html page that auto-generates documentation based on the JSDoc comments

## Example
<img width="237" height="332" alt="image" src="https://github.com/user-attachments/assets/6f96f712-dba3-459b-aa7c-f4f215c69bf2" />

```
section | Pills
pills_start
pill | Red pill | red
pill | Blue pill | blue
pill | Green pill | green
pill | Orange pill | orange
pill | Teal pill | teal
pill | Goldenrod pill | goldenrod
pill | Hotpink pill | hotpink
pill | Grey pill | grey
pill | Black pill | black
pills_end
fill | 1
section | Table
table_start
table_head | Level | Loc | Range | … | Cost
table_row | 2 | City | 5 | … | 100
table_row | 6 | Hamlet | 2 | … | 1000
table_row | 17 | Cave | 50 | … | 10
table_end
fill | 1
section | Italic
italic | This text is in italics
fill | 1
section | SR6spell
sr6spell | Touch | P | S | 5 | 2
```

### Collaborators
@tgreatdk

google-labs-jules[bot] committed:
feat: Add GitHub Action for building and deploying to GitHub Pages

bnm12 committed:
Merge pull request #1 from bnm12/feat/add-github-action

feat: Add GitHub Action for building and deploying to GitHub Pages

google-labs-jules[bot] committed:
I will now implement the `sr6spell` block.

bnm12 committed:
Merge pull request #2 from bnm12/sr6spell-block

I will now implement the `sr6spell` block.

bnm12 committed:
Update deploy.yml

bnm12 committed:
Update deploy.yml

bnm12 committed:
Update deploy.yml

google-labs-jules[bot] committed:
feat: Implement html action

This commit introduces a new `html` action that allows rendering raw HTML content on cards. A new `card_element_html` function has been added to handle the rendering, and it has been registered in the `card_element_generators` object.

bnm12 committed:
Merge pull request #3 from bnm12/feat/html-action

feat: Implement html action

bnm12 committed:
Update ui.css

bnm12 committed:
Update cards.css

bnm12 committed:
Update cards.js

google-labs-jules[bot] committed:
I am now working on implementing the card action for 'table'.

bnm12 committed:
Merge pull request #4 from bnm12/feature/table-action

I am now working on implementing the card action for 'table'.

google-labs-jules[bot] committed:
Added a list of card actions.

bnm12 committed:
Merge pull request #5 from bnm12/feature/card-actions-list

Added a list of card actions.

google-labs-jules[bot] committed:
I am adding a heading and styling to the actions list.

bnm12 committed:
Merge pull request #6 from bnm12/actions-list-styling

I am adding a heading and styling to the actions list.

bnm12 committed:
Update cards.js

bnm12 committed:
Update ui.js

google-labs-jules[bot] committed:
Feat: Group card actions and display as buttons

bnm12 committed:
Merge pull request #7 from bnm12/feature/grouped-actions

Feat: Group card actions and display as buttons

google-labs-jules[bot] committed:
Fix: Append to content box instead of clearing it

bnm12 committed:
Merge pull request #8 from bnm12/fix/append-to-content

Fix: Append to content box instead of clearing it

google-labs-jules[bot] committed:
Fix: Prevent action buttons from reloading page

bnm12 committed:
Merge pull request #9 from bnm12/fix/button-reload

Fix: Prevent action buttons from reloading page

Commit:
Add margin to white card

bnm12 committed:
Merge pull request #10 from bnm12/fix/card-align

Add margin to white card

google-labs-jules[bot] committed:
I am going to format the JSDoc comments.

bnm12 committed:
Merge pull request #11 from bnm12/feat/format-jsdoc-comments

I am going to format the JSDoc comments.

google-labs-jules[bot] committed:
I added tooltips and examples to the action buttons. To do this, I created a JSDoc parser to extract summaries and examples from `cards.js` and modified the UI to display the summary as a tooltip on hover and insert the example text on click.

bnm12 committed:
Merge pull request #12 from bnm12/feature/jsdoc-tooltips

I added tooltips and examples to the action buttons. To do this, I cr…

bnm12 committed:
Update cards.js with table documentation

google-labs-jules[bot] committed:
I am fixing the card action lookup to use the function name.

bnm12 committed:
Merge pull request #13 from bnm12/feature/lookup-action-by-function-name

I am fixing the card action lookup to use the function name.

google-labs-jules[bot] committed:
Refactor: Move card action categories to JSDoc

bnm12 committed:
Merge pull request #14 from bnm12/feature/dynamic-card-actions

Refactor: Move card action categories to JSDoc

google-labs-jules[bot] committed:
feat: Implement and document the italic action

bnm12 committed:
Merge pull request #15 from bnm12/feature/italic-action

feat: Implement and document the italic action

google-labs-jules[bot] committed:
feat: update and expand sample cards

bnm12 committed:
Merge pull request #16 from bnm12/feat/comprehensive-sample-cards

feat: update and expand sample cards

google-labs-jules[bot] committed:
I am creating the documentation page for card actions.

bnm12 committed:
Merge pull request #17 from bnm12/feature/documentation-page

I am creating the documentation page for card actions.

google-labs-jules[bot] committed:
Fix: Update hardcoded URLs to relative paths

- I updated README.md to point to the new documentation and preview URLs.
- I updated generator/index.html to use a relative path for the documentation link.

bnm12 committed:
Merge pull request #18 from bnm12/fix/update-links

Fix: Update hardcoded URLs to relative paths

google-labs-jules[bot] committed:
Fix empty card layout and add classes

google-labs-jules[bot] committed:
Fix empty card layout and add classes

bnm12 committed:
Merge pull request #19 from bnm12/feature/empty-card-layout-fix

Fix empty card layout and add classes

bnm12 commented:
How do you feel about a PR of this size @mephitrpg ?

google-labs-jules[bot] committed:
feat: Add pills action group

This commit adds a new card action group for pills, similar to the table and p2e_trait action groups.

The new action group is used as follows:

pills_start
pill | html-color | text
pills_end

bnm12 committed:
Merge pull request #20 from bnm12/feat/pills-action-group

feat: Add pills action group

google-labs-jules[bot] committed:
feat: Add bootstrap pill styling to card action

- Adjusts the pill action to have more of a default bootstrap pill styling (use bootstrap classes if possible) with rounded corners and a bit of padding.
- Makes the text color contrast the background color

bnm12 committed:
Merge pull request #21 from bnm12/feature/bootstrap-pill-styling

feat: Add bootstrap pill styling to card action

google-labs-jules[bot] committed:
Fix: Apply Bootstrap pill styling in card.js

The pill action in card.js was not applying the Bootstrap pill styling correctly. This was because the HTML was not structured correctly for Bootstrap pills.

This commit fixes the issue by:
- Wrapping the pills in a `ul` with the `nav` and `nav-pills` classes.
- Wrapping each pill in an `li` element.
- Removing unnecessary classes from the pill element.

bnm12 committed:
Merge pull request #22 from bnm12/fix/pill-styling

Fix: Apply Bootstrap pill styling in card.js

google-labs-jules[bot] committed:
I've addressed the issue with the pill button's partial matching. I've removed the name splitting from `docparser` and `ui.js` and implemented full method name matching instead.

bnm12 committed:
Merge pull request #23 from bnm12/fix-pill-button-partial-match

I've addressed the issue with the pill button's partial matching. I'v…

google-labs-jules[bot] committed:
Rework pills to use bootstrap badge pills

bnm12 committed:
Merge pull request #24 from bnm12/feature/bootstrap-badge-pills

Rework pills to use bootstrap badge pills

google-labs-jules[bot] committed:
Fix(cards): Update pills to use Bootstrap v3 classes

google-labs-jules[bot] committed:
Fix(cards): Update pills to use Bootstrap v3 classes

bnm12 committed:
Merge pull request #25 from bnm12/feat/update-bootstrap-v3-classes

Feat/update bootstrap v3 classes

google-labs-jules[bot] committed:
Fix: Switch pill action argument order

bnm12 committed:
Merge pull request #26 from bnm12/feature/pill-argument-switch

Fix: Switch pill action argument order

bnm12 committed:
Fix overflow in card

bnm12 committed:
Update output.html

bnm12 committed:
Update ui.css

bnm12 committed:
Bold description and property keye

bnm12 committed:
Update card_data_example.js

bnm12 committed:
Update pills colors and style

bnm12 committed:
Update cards.css

bnm12 committed:
Update cards.js

bnm12 committed:
Update cards.js

bnm12 committed:
Update cards.css

bnm12 committed:
Update cards.css

bnm12 committed:
Add files via upload

bnm12 committed:
Move bootstrap files

bnm12 committed:
Update index.html

bnm12 committed:
No more -webkit-

bnm12 committed:
Attempt to align card fronts with back bleed

bnm12 committed:
Add white card margins the correct way

bnm12 committed:
Set back bleed to 0mm default

bnm12 committed:
Add margin to whilte cards

bnm12 committed:
Fix empty cards

mephitrpg commented:
Hi, this is a great enhancement, thanks!
About documentation.html and "Deploy using github pages rather than Vercel". Does the documentation actually published in the github pages will be completely replaced by the application?

bnm12 commented:
> Hi, this is a great enhancement, thanks! About documentation.html and "Deploy using github pages rather than Vercel". Does the documentation actually published in the github pages will be completely replaced by the application?

You can see how it works by going to the hosted link. But it's not a 1:1 replacement

mephitrpg commented:
Thank you! I merged some other pull requests today — please resolve the conflicts so I can merge yours as well.

gitbutler-client committed:
GitButler Workspace Commit

This is placeholder commit and will be replaced by a merge of your virtual branches.

Due to GitButler managing multiple virtual branches, you cannot switch back and
forth between git branches and virtual branches easily. 

If you switch to another branch, GitButler will need to be reinitialized.
If you commit on this branch, GitButler will throw it away.

For more information about what we're doing here, check out our docs:
https://docs.gitbutler.com/features/virtual-branches/integration-branch

bnm12 committed:
Merge remote-tracking branch 'upstream/master' into gitbutler/workspace

bnm12 committed:
Merge pull request #27 from bnm12/feat/merge-upstream

Merge upstream

bnm12 commented:
Sorry I was busy for a bit there, I have have fixed the conflicts now @mephitrpg

mephitrpg commented:
Thank you @bnm12
